### PR TITLE
Issue 1 - Fix OpenMP crash on MacOS

### LINKED ISSUE #1: Error with OpenMP
### Informations

- **Qiskit AER version**:
qiskit 0.1
- **Operating system**:
MacOS 

### What is the current behavior?
Running the tutorials gives the following error

OMP: Error #15: Initializing libiomp5.dylib, but found libomp.a already initialized.
OMP: Hint This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results. For more information, please see http://www.intel.com/software/products/support/.

### What is the expected behavior?
No error

### PR DESCRIPTION
**Pull request by atilag**
### Summary
Fixes #1 

### Details and comments
On Mac, OpenMP implementation is integrated through an external library: `libomp.dylib` (and some others similar like libiomp5 from Intel). This library has a constraint: It cannot be loaded twice in memory, basically because it will imply a double initialization of it's runtime, and this is potentially dangerous. We do need this library to operate, but other dependencies we have, require them as well, which is a problem because of the way Python loads extensions (and we are a Python extension too).
Basically, with this fix, we have provided Aer the ability to dynamically load libomp in every possible scenario without incurring in the double initialization problem.

### CODE REVIEW

**Comment by chriseclectic** [comment]
@atilag Do you know whats going on with the tests? This PR is resulting in several thousand lines of build warnings/errors in Travis and all the tests erroring out with

```
ModuleNotFoundError: No module named 'qiskit.providers.aer.backends.qasm_controller_wrapper'
```

Locally I built with XCode and GCC8 on MacOS and the `aer_provider` tutorial worked both in a pip installed Numpy env, and a conda installed MLK numpy env, but the tests are all failing locally with the same error as Travis (but without all the compiler warnings).

**Comment by atilag** [comment]
Yeah, I know what`s going on. Fix is comming...

**Comment by atilag** [comment]
Everything seems to be working now :) , isn't it @chriseclectic ?

**Comment by chriseclectic** [comment]
Thanks @atilag, you deserve a beer or 6 after that!
